lib/strings.nix

```diff
@@ -305,6 +330,18 @@ rec {
   */
   escape = list: replaceChars list (map (c: "\\${c}") list);
 
+  /* Escape occurence of the element of `list` in `string` by
+     converting to it's ASCII value and prefixing it with \\x.
```

Comment:
mention that it only works with printable ascii characters. For non printable ones, I wonder if it is correct to map the caracter 7 to `\x7` instead of `\x07`.

Reply:
I've now documented that it only works for printable characters, You're quite right about the non-printable ones as well, but I think we would need a nix builtin to deal with those anyways